.gitignore

```diff
@@ -1,25 +1,31 @@
-/target/
-!.mvn/wrapper/maven-wrapper.jar
+*.iml
+.idea
+target/
+build/
+.gradle/
+pom.xml.tag
+pom.xml.releaseBackup
+pom.xml.versionsBackup
+pom.xml.next
+release.properties
+dependency-reduced-pom.xml
+buildNumber.properties
+.mvn/timing.properties
 
-### STS ###
-.apt_generated
-.classpath
-.factorypath
-.project
+.cache
+
+# Avoid ignoring Maven wrapper jar file (.jar files are usually ignored)
+!/.mvn/wrapper/maven-wrapper.jar
 .settings
-.springBeans
-.sts4-cache
+.vscode
+.project
+.classpath
+docs/output/*
+/hello.txt
+.idea/
+work/**/*
+!work/README.md
+#
 
-### IntelliJ IDEA ###
-.idea
-*.iws
-*.iml
-*.ipr
+docker-build.yaml
 
-### NetBeans ###
-/nbproject/private/
-/build/
-/nbbuild/
-/dist/
-/nbdist/
```

Comment:
@lance - don’t remove all these, I have settings for all common IDEs that a developer uses

Reply:
@kameshsampath I simply rebased from `master`. Those entries do not exist there: https://github.com/redhat-developer-demos/knative-tutorial-greeter/blob/master/.gitignore